### PR TITLE
[Release Guide] Feature Design

### PR DESCRIPTION
**Pull request by mkolesnik**
Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

### CODE REVIEW

File: src/content/development/contribution-guide/_index.en.md [hunk 1]
```diff
@@ -62,7 +62,35 @@ To be expanded
...
+
+In case additional work is identified during the design, it should be opened as [tasks](#tasks) and tracked under the respective epic.
+Such tasks are expected to follow the sizing guidelines from the [Pre-Planning](#pre-planning) stage.
+Specifically, tasks that are themselves epics due to their size should be identified and treated as such.
+
+Design proposals for epics should be submitted as [pull requests to the enhancements repository], detailing the proposed design, any
+alternatives, and any changes necessary to the Submariner projects and APIs.
+
+The [pull requests to the enhancements repository] will be reviewed during the sprint, discussing any necessary changes or reservations.
+Any pull request will need approval from at least 50% of the [code owners of the enhancements repository].
```
Comment (skitt) [inline src/content/development/contribution-guide/_index.en.md:150]: Should we involve the code owners of the main “target” repository? If so, how do we determine the main target repository?

Incidentally, the list of enhancements code owners was outdated; see https://github.com/submariner-io/enhancements/pull/86

File: src/content/development/contribution-guide/_index.en.md [hunk 2]
```diff
@@ -62,7 +62,35 @@ To be expanded
...
+
+Project members are expected to utilize the sprint to draft their own proposals, and review proposals from other participants.
+If a project member has finished work on their proposal, they're encouraged to help with the other ongoing proposals.
+
+Only epics which were planned for the release will be reviewed at this stage.
+Any epic that was unplanned but seeks inclusion in the release should follow the [exception process](#exception-process).
+The same process will have to be followed for any epic that was planned but has not been agreed upon by the end of this sprint.
+Any such epics will be moved back to the [Backlog board](#backlog-board) and reconsidered for the release.
+
+At the end of the sprint, a special release `m0` signifying the 0th milestone will be released.
```
Comment (skitt) [inline src/content/development/contribution-guide/_index.en.md]: Do we have to wait for the end of the sprint? The idea behind m0 was to follow the previous release immediately, to avoid situations such as the `go.mod` bumps in https://github.com/submariner-io/submariner-operator/pull/1961

**Review by skitt — CHANGES_REQUESTED** [review]

File: src/content/development/contribution-guide/_index.en.md [hunk 3]
```diff
@@ -62,7 +62,35 @@ To be expanded
 
 #### Feature Design
 
-To be expanded
+Project members are expected to work on feature design for any [epic](#epics) feature assigned to them.
+During this sprint, project members will update their respective epics with any work identified during the design page.
```
Comment (Jaanki) [inline src/content/development/contribution-guide/_index.en.md]:
```suggestion
During this sprint, project members will update their respective epics with any work identified during the design phase.
```
Maybe you meant phase?

File: src/content/development/contribution-guide/_index.en.md (hunk 2 above)
Comment (Jaanki) [reply src/content/development/contribution-guide/_index.en.md]: m0  is when planning/designing is happening. No code goes in. Just getting enhancement proposals merged. Dependency bumps are maintenance work and can be done any time through the release cycle.

**Review by Jaanki — CHANGES_REQUESTED** [review]

File: src/content/development/contribution-guide/_index.en.md (hunk 2 above)
Comment (skitt) [reply src/content/development/contribution-guide/_index.en.md]: My point is that by leaving m0 to the end of the first sprint, we run the risk of requiring contributors to bump unrelated dependencies, as happened in submariner-io/submariner-operator#1961. The release branches are kept in sync during the release, but that’s not true of the `devel` branch once the release branch is created. Having an m0 as soon as possible reduces the risk, even if we forbid code changes during the planning sprint (which seems unrealistic to me, given the need to fix the release by backporting fixes from the `devel` branches).

File: src/content/development/contribution-guide/_index.en.md [hunk 4]
```diff
@@ -62,7 +62,35 @@ To be expanded
...
+Any pull request will need approval from at least 50% of the [code owners of the enhancements repository].
+When the pull request is reviewed and merged, work on the epic can begin.
+
+Project members are expected to utilize the sprint to draft their own proposals, and review proposals from other participants.
+If a project member has finished work on their proposal, they're encouraged to help with the other ongoing proposals.
+
+Only epics which were planned for the release will be reviewed at this stage.
+Any epic that was unplanned but seeks inclusion in the release should follow the [exception process](#exception-process).
+The same process will have to be followed for any epic that was planned but has not been agreed upon by the end of this sprint.
+Any such epics will be moved back to the [Backlog board](#backlog-board) and reconsidered for the release.
```
Comment (sridhargaddam) [inline src/content/development/contribution-guide/_index.en.md]:
```suggestion
Any such epics will be moved back to the [Backlog board](#backlog-board) and reconsidered for the next release.
```

File: src/content/development/contribution-guide/_index.en.md (hunk 2 above)
Comment (mkolesnik) [reply src/content/development/contribution-guide/_index.en.md]: I suggest we tie m0 to the end of this sprint as it makes "conceptual sense" and isn't an outlier, related to the rest of the sprints.

If we see it causes too much problems I suggest we raise an issue (or PR) to change the release process to have m0 right after GA and then we'll update the doc.

Comment (skitt) [reply src/content/development/contribution-guide/_index.en.md]: The thing is, we know it would have saved us some time at the start of the 0.11 cycle if we’d done an m0 then; we did a 0.12 m0 right after 0.11 GA, which was useful; and it would have saved us some time and confusion if we’d done an m0 for 0.13 right after 0.12 GA. We’ve run into this problem in the last three releases at least, I suspect we’ll encounter it again... I know it means m0 comes out of step compared to m1/m2 etc., but I asked for it in 0.12 because it fixes an actual problem, not because it looks nice on the timeline.

Comment (dfarrell07) [reply src/content/development/contribution-guide/_index.en.md]: I think what makes "conceptual sense" can be adjusted to match what works best if we think about/explain the reasoning behind it. It makes conceptual sense to me, especially after Stephen's explanation, that  for `m0` comes at branch cutting and all the other `m` values are named `mn+1` (or there isn't a milestone release at the end of the planning sprint).

File: src/content/development/contribution-guide/_index.en.md (hunk 1 above)
Comment (dfarrell07) [reply src/content/development/contribution-guide/_index.en.md:150]: Yeah, I think it makes sense that the code owners of the code that is expected to change are the ones that should vote.

File: src/content/development/contribution-guide/_index.en.md [hunk 5]
```diff
@@ -62,7 +62,35 @@ To be expanded
 
 #### Feature Design
 
-To be expanded
+Project members are expected to work on feature design for any [epic](#epics) feature assigned to them.
```
Comment (dfarrell07) [inline src/content/development/contribution-guide/_index.en.md]:
```suggestion
Project members are expected to work on feature design for any [epic](#epics) features assigned to them.
```

Comment (tpantelis) [inline src/content/development/contribution-guide/_index.en.md]:
```suggestion
Project members are expected to work on the design for any [epic](#epics) features assigned to them.
```

File: src/content/development/contribution-guide/_index.en.md [hunk 6]
```diff
@@ -62,7 +62,35 @@ To be expanded
...
+During this sprint, project members will update their respective epics with any work identified during the design phase.
```
Comment (tpantelis) [inline src/content/development/contribution-guide/_index.en.md:139]: What sprint? It mentions "this sprint" w/o previously defining it. Also should we confine feature design  to a specific sprint or time frame? I'd think feature design could be done anytime.

File: src/content/development/contribution-guide/_index.en.md [hunk 7]
```diff
@@ -62,7 +62,35 @@ To be expanded
...
+Project members are encouraged to perform "proof of concept" investigations in order to consolidate the design and understand any
+necessary work items.
```
Comment (tpantelis) [inline src/content/development/contribution-guide/_index.en.md]: I don't think _proof of concept_ needs to be in qoutes.

```suggestion
Project members are encouraged to perform proof of concept investigations in order to validate the design and clarify specific work items.
```

File: src/content/development/contribution-guide/_index.en.md [hunk 8]
```diff
@@ -62,7 +62,35 @@ To be expanded
...
+
+In case additional work is identified during the design, it should be opened as [tasks](#tasks) and tracked under the respective epic.
```
Comment (tpantelis) [inline src/content/development/contribution-guide/_index.en.md]:
```suggestion
In case additional work items are identified during the design, they should be opened as [tasks](#tasks) and tracked under the respective epic.
```

File: src/content/development/contribution-guide/_index.en.md [hunk 9]
```diff
@@ -62,7 +62,35 @@ To be expanded
...
+
+Design proposals for epics should be submitted as [pull requests to the enhancements repository], detailing the proposed design, any
+alternatives, and any changes necessary to the Submariner projects and APIs.
+
+The [pull requests to the enhancements repository] will be reviewed during the sprint, discussing any necessary changes or reservations.
+Any pull request will need approval from at least 50% of the [code owners of the enhancements repository].
+When the pull request is reviewed and merged, work on the epic can begin.
+
+Project members are expected to utilize the sprint to draft their own proposals, and review proposals from other participants.
+If a project member has finished work on their proposal, they're encouraged to help with the other ongoing proposals.
```
Comment (tpantelis) [inline src/content/development/contribution-guide/_index.en.md]:
```suggestion
Project members are expected to review proposals from other members in addition to drafting their own proposals.
If a project member has finished work on their proposal, they're encouraged to help with the other ongoing proposals.
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: src/content/development/contribution-guide/_index.en.md (hunk 6 above)
Comment (mkolesnik) [reply src/content/development/contribution-guide/_index.en.md:139]: This is part or the timeline section which defines all the sprints, so each sub section is a sprint.

The limit is on content yes, anything not done by the end of the sprint has no guarantee for attention after it and will be punted to the next version (unless its accepted through the exception process)

File: src/content/development/contribution-guide/_index.en.md (hunk 1 above)
Comment (mkolesnik) [reply src/content/development/contribution-guide/_index.en.md:150]: TBH I think it's easier to track one list which is also in the same repo and adds a shield icon in the UI rather than separate multiple lists.
We can expand the code owners list on enhancements to include the code owners from the other repos, since usually EPs on enhancements tend to affect multiple repos.

TL;DR let's keep it simple, lets expand codeowners of enhancements

File: src/content/development/contribution-guide/_index.en.md (hunk 2 above)
Comment (mkolesnik) [reply src/content/development/contribution-guide/_index.en.md]: We can have this one be floating if you like and I can capture it on the doc, but ATM is there a need for 0.13.0-m0?

> and it would have saved us some time and confusion if we’d done an m0 for 0.13 right after 0.12 GA

What's the problem?

Comment (skitt) [reply src/content/development/contribution-guide/_index.en.md]: The problem is that at the start of a release cycle, the main development branches aren’t necessarily in sync, and that can mean that changes across multiple repositories can require more work than strictly necessary. This ends up being confusing for new contributors. https://github.com/submariner-io/submariner-operator/pull/1961 is the main example in this cycle but we’ve had similar issues in previous cycles (which is why I’d asked for an m0 previously).

This time round, some framework changes went into Shipyard, but the necessary changes weren’t propagated throughout the projects. For https://github.com/submariner-io/submariner-operator/pull/1961, an unrelated change needed to be made in Shipyard, and pulled into the Operator. The resulting version bump failed because it meant pulling in all of the recent changes to Shipyard, which broke the version of Lighthouse that the Operator depended upon. The fix was to _also_ bump Lighthouse even though it has nothing to do with this change. This is perhaps obvious to us, but we can’t expect all contributors to have to deal with this sort of issue. Because our release branches are cut early (and I’m _not_ saying we should revisit that), and dependencies are aligned by the release process, this doesn’t show up in our release branches but leaves plenty of time between the branch cut and the start of the next cycle for it to show up in the development branch.

Comment (skitt) [reply src/content/development/contribution-guide/_index.en.md]: So in 0.13, right now, we don’t need this change because it’s been worked around.

There’s also the issue that currently, the development branches have older versions than the release branches (from a Go module versioning perspective), which means that in some circumstances `go.mod` handling can “upgrade” the development branch to a release tag. Having an m0 also avoids this.

Comment (skitt) [reply src/content/development/contribution-guide/_index.en.md]: Given that this is the fourth or fifth time I’ve explained this, perhaps we can ignore this requirement in this PR and I’ll write a detailed PR, with the explanations, that way we won’t forget the reasoning behind m0 (assuming we agree that m0 at the very start of the cycle is a good idea).

**Review by tpantelis — APPROVED** [review]

File: src/content/development/contribution-guide/_index.en.md (hunk 2 above)
Comment (mkolesnik) [reply src/content/development/contribution-guide/_index.en.md]: So for the example you gave https://github.com/submariner-io/submariner-operator/pull/1961 I don't quite understand why a m0 would help?
Let's say we had cut m0 at some point, and then the changes to Shipyard happened, wouldn't then you need to bump the versions as was done regardless?

I think the m0 in this scenario is a red herring as the projects are inter-connected and thus once you introduce some feature in a library project (shipyard or admiral) you'd have to sync the version across the consuming project in order to introduce a change to operator, since operator ties everything in. This has really nothing to do with m0 or any other tag, its just a byproduct of how our projects depend on each other.

Also based on your explained logic, once the stable branch is cut realistically the stable branch could "run ahead" of the devel branch at any point and especially rcs so wouldnt it be better to release m0 right after rc0, to avoid any such issue (go.mod auto-updating)?

The idea of a separate PR could make sense for the clarity in the git log why this is the way it is, otherwise it's all in the github.

Comment (skitt) [reply src/content/development/contribution-guide/_index.en.md]: > So for the example you gave [submariner-io/submariner-operator#1961](https://github.com/submariner-io/submariner-operator/pull/1961) I don't quite understand why a m0 would help? Let's say we had cut m0 at some point, and then the changes to Shipyard happened, wouldn't then you need to bump the versions as was done regardless?

Yes, and the point of m0 there is to ensure that this does happen.

> Also based on your explained logic, once the stable branch is cut realistically the stable branch could "run ahead" of the devel branch at any point and especially rcs so wouldnt it be better to release m0 right after rc0, to avoid any such issue (go.mod auto-updating)?

That’s what I asked for originally, when we first cut an m0. I can’t remember who argued against doing it then back then, and why we’re considering m0 later now; I’m just trying to avoid having m0 slip even further.

> The idea of a separate PR could make sense for the clarity in the git log why this is the way it is, otherwise it's all in the github.

Who reads PR comments after a PR has been merged?

Comment (mkolesnik) [reply src/content/development/contribution-guide/_index.en.md]: > > So for the example you gave [submariner-io/submariner-operator#1961](https://github.com/submariner-io/submariner-operator/pull/1961) I don't quite understand why a m0 would help? Let's say we had cut m0 at some point, and then the changes to Shipyard happened, wouldn't then you need to bump the versions as was done regardless?
> 
> Yes, and the point of m0 there is to ensure that this does happen.
> 

I think you missed my point?
If m0 happens right after the rc0, then some time later that library change happens, versions would have to be synced anyway.
So m0 wont really help this "proble" which I'd argue isn't a problem since it's just how our projects dependencies are structured.

> 
> That’s what I asked for originally, when we first cut an m0. I can’t remember who argued against doing it then back then, and why we’re considering m0 later now; I’m just trying to avoid having m0 slip even further.
> 

OK then lets discuss this on a broader forum and of course update the docs when we reach a conclusion. For now we can merge this and update later since this isn't the ten commandments ;)

> 
> Who reads PR comments after a PR has been merged?

Only a special type of person ;)

Comment (skitt) [reply src/content/development/contribution-guide/_index.en.md]: > > > So for the example you gave [submariner-io/submariner-operator#1961](https://github.com/submariner-io/submariner-operator/pull/1961) I don't quite understand why a m0 would help? Let's say we had cut m0 at some point, and then the changes to Shipyard happened, wouldn't then you need to bump the versions as was done regardless?
> > 
> > 
> > Yes, and the point of m0 there is to ensure that this does happen.
> 
> I think you missed my point? If m0 happens right after the rc0, then some time later that library change happens, versions would have to be synced anyway. So m0 wont really help this "proble" which I'd argue isn't a problem since it's just how our projects dependencies are structured.

No, I didn’t miss your point. My point is that, from project experience, dependency discrepancy tends to come to a head at release time (and yes, perhaps we should encourage processes to avoid that). That discrepancy is resolved in the release branches by the release process, but we don’t have any guaranteed resolution in the development branch. I _know_ having m0 at a fixed point doesn’t guarantee we’ll fix all such occurrences all the time; if we could avoid such occurrences all the time, we wouldn’t need to fix them.

Having m0 just after rc0 ensures that, at that point in time, both the release branch and the development branch are consistent across all projects. For the development branch it also has the nice side effect of ensuring that the development branch always has a greater version than the latest release branch.

> > That’s what I asked for originally, when we first cut an m0. I can’t remember who argued against doing it then back then, and why we’re considering m0 later now; I’m just trying to avoid having m0 slip even further.
> 
> OK then lets discuss this on a broader forum and of course update the docs when we reach a conclusion. For now we can merge this and update later since this isn't the ten commandments ;)

We’ve been discussing this, including in broader forums, for the last three cycles. I don’t think bringing it up _again_ is going to be productive.

> > Who reads PR comments after a PR has been merged?
> 
> Only a special type of person ;)

Which is my point — it’s better to have this information in the documentation or in commit messages.

File: src/content/development/contribution-guide/_index.en.md [hunk 10]
```diff
@@ -62,7 +62,34 @@ To be expanded
 
 #### Feature Design
 
-To be expanded
+Project members are expected to work on the design for any [epic](#epics) features assigned to them.
+During this sprint, project members will update their respective epics with any work identified during the design phase.
+Project members are encouraged to perform proof of concept investigations in order to validate the design and clarify specific work items.
+
+In case additional work items are identified during the design, they should be opened as [tasks](#tasks) and tracked under the respective epic.
+Such tasks are expected to follow the sizing guidelines from the [Pre-Planning](#pre-planning) stage.
```
Comment (maayanf24) [inline src/content/development/contribution-guide/_index.en.md]:
```suggestion
Such tasks are expected to follow the sizing guidelines from the [Planning](#planning) stage.
```

File: src/content/development/contribution-guide/_index.en.md [hunk 11]
```diff
@@ -62,7 +62,34 @@ To be expanded
...
+Specifically, tasks that are themselves epics due to their size should be identified and treated as such.
+
+Design proposals for epics should be submitted as [pull requests to the enhancements repository], detailing the proposed design, any
+alternatives, and any changes necessary to the Submariner projects and APIs.
+
+The [pull requests to the enhancements repository] will be reviewed during the sprint, discussing any necessary changes or reservations.
+Any pull request will need approval from at least 50% of the [code owners of the enhancements repository].
+When the pull request is reviewed and merged, work on the epic can begin.
```
Comment (maayanf24) [inline src/content/development/contribution-guide/_index.en.md]: Maybe add that there is no need to wait to the end of the design sprint, and work on an epic can begin once the enhance PR is merged

File: src/content/development/contribution-guide/_index.en.md (hunk 2 above)
Comment (dfarrell07) [reply src/content/development/contribution-guide/_index.en.md]: +1 to m0 at the start of the cycle and a dedicated PR to explain it

File: src/content/development/contribution-guide/_index.en.md (hunk 11 above)
Comment (mkolesnik) [reply src/content/development/contribution-guide/_index.en.md]: This is exactly what this line is saying, do you have a better way to say it?

It seems to me "work on the epic can begin" is pretty straightforward but I might be wrong

File: src/content/development/contribution-guide/_index.en.md (hunk 10 above)
Comment (mkolesnik) [reply src/content/development/contribution-guide/_index.en.md]: This wasn't merged yet and the name in the skeleton is "Pre-Planning"

File: src/content/development/contribution-guide/_index.en.md (hunk 2 above)
Comment (mkolesnik) [reply src/content/development/contribution-guide/_index.en.md]: This doesn't seem to progress here, let's track it on #738 and continue with PR as is for now until theres a consensus and we can update the doc with the result

Comment (dfarrell07) [reply src/content/development/contribution-guide/_index.en.md]: Can we change this PR to reflect that there's not consensus? As-is, it reflects your position.

File: src/content/development/contribution-guide/_index.en.md (hunk 10 above)
Comment (skitt) [reply src/content/development/contribution-guide/_index.en.md]: The skeleton can be wrong ;-). The purpose of the “Pre-Planning stage” as described isn’t pre-planning, it’s planning — pre-planning is ahead-of-time definition of a set of planned steps, which is actually what we’re doing in these PRs; it’s not what happens in the development stage described here (that’s just “planning”).

File: src/content/development/contribution-guide/_index.en.md [hunk 12]
```diff
@@ -62,7 +62,34 @@ To be expanded
...
+
+Project members are expected to review proposals from other members in addition to drafting their own proposals.
+If a project member has finished work on their proposal, they're encouraged to help with the other ongoing proposals.
+
+Only epics which were planned for the release will be reviewed at this stage.
+Any epic that was unplanned but seeks inclusion in the release should follow the [exception process](#exception-process).
+The same process will have to be followed for any epic that was planned but has not been agreed upon by the end of this sprint.
+Any such epics will be moved back to the [Backlog board](#backlog-board) and reconsidered for the next release.
+
+At the end of the sprint, a special release `m0` signifying the 0th milestone will be released.
```
Comment (skitt) [inline src/content/development/contribution-guide/_index.en.md]:
```suggestion
```

since there’s no consensus.

**Review by skitt — CHANGES_REQUESTED** [review]

File: src/content/development/contribution-guide/_index.en.md (hunk 11 above)
Comment (dfarrell07) [reply src/content/development/contribution-guide/_index.en.md]: I agree that the current version is clear, but maybe something like:

> As soon as the enhancement pull request is reviewed and merged, work on the epic can begin.

Would emphasize @maayanf24's point that there's no need to wait until the end of the sprint.

File: src/content/development/contribution-guide/_index.en.md (hunk 10 above)
Comment (mkolesnik) [reply src/content/development/contribution-guide/_index.en.md]: I'm not saying it's not wrong, but this will lead to CI breakage.

I'm aware of this and whatever PR lands second will have to adjust the link anyhow, but right now this passes CI and the suggested change won't

File: src/content/development/contribution-guide/_index.en.md (hunk 1 above)
Comment (Jaanki) [reply src/content/development/contribution-guide/_index.en.md:150]: Agree, lets expand codeowners of enhancements to include code owners from other repos and core submariner team (regular contributors to the project)

File: src/content/development/contribution-guide/_index.en.md (hunk 10 above)
Comment (skitt) [reply src/content/development/contribution-guide/_index.en.md]: OK, let’s leave it as-is and fix everything in one shot later.

**Review by skitt — APPROVED** [review]

File: src/content/development/contribution-guide/_index.en.md (hunk 1 above)
Comment (dfarrell07) [reply src/content/development/contribution-guide/_index.en.md:150]: If we go that route, should we document that the code owners of the enhancements repos should be the set of all code owners of SubM code?

**Review by maayanf24 — APPROVED** [review]

**Review by dfarrell07 — CHANGES_REQUESTED** [review]
Properly marking -1 so we don't miss the inline discussion about documenting the changes to enhancement repo owners this necessitates.

File: src/content/development/contribution-guide/_index.en.md (hunk 1 above)
Comment (mkolesnik) [reply src/content/development/contribution-guide/_index.en.md:150]: Should this be part of this documentation though?

Comment (dfarrell07) [reply src/content/development/contribution-guide/_index.en.md:150]: I think so? It's about agreeing on features, about this new planning process. I don't think it should go in https://submariner.io/community/contributor-roles/, the only other relevant place I can think of, as it's not really part of the core Member/Owners design.

Comment (mkolesnik) [reply src/content/development/contribution-guide/_index.en.md:150]: If it's agreed upon then I'll add it to this section

The proposed line is here: https://github.com/submariner-io/submariner-website/pull/721/commits/413baebf3992a1587e41384c4e97fb59ae9ad739

**Review by dfarrell07 — APPROVED** [review]

File: src/content/development/contribution-guide/_index.en.md [hunk 13]
```diff
@@ -62,7 +62,32 @@ To be expanded
...
+
+The [pull requests to the enhancements repository] will be reviewed during the sprint, discussing any necessary changes or reservations.
+Any pull request will need approval from at least 50% of the [code owners of the enhancements repository].
+The code owners list is an aggregate list of the code owners of all Sumbariner repositories.
+As soon as the pull request is reviewed and merged, work on the epic can begin.
+
+Project members are expected to review proposals from other members in addition to drafting their own proposals.
+If a project member has finished work on their proposal, they're encouraged to help with the other ongoing proposals.
+
+Only epics which were planned for the release will be reviewed at this stage.
```
Comment (dfarrell07) [inline src/content/development/contribution-guide/_index.en.md:157]: To be clear, approving and merging these docs does not necessarily reflect an official, enforceable policy of Submariner. Submariner's Governance gives Project Owners the ultimate responsibility to "Set technical direction and priorities for the project" and "Define milestones and releases".

https://submariner.io/community/contributor-roles/#owner-responsibilities-and-privileges

Comment (Jaanki) [reply src/content/development/contribution-guide/_index.en.md:157]: This is not true. Everything that gets merged, happens **only after consensus** from some portion of the team.

Comment (dfarrell07) [reply src/content/development/contribution-guide/_index.en.md:157]: I'm not sure I know what you mean? Let me give an example. Imagine you were to send a PR creating docs that "Janki decides everything regarding planning", I as a website Owner +2 it, some non-Owner +1s, and we merge it. That would not be an enforceable Submariner policy, because our governance gives Project Owners that responsibility, not the three of us.

Comment (dfarrell07) [reply src/content/development/contribution-guide/_index.en.md:157]: To provide context for folks who haven't attended recent Submariner calls: this line is being cited as a reason we must follow this course of action, as if it's already agreed-on authoritative doctrine.

Comment (mkolesnik) [reply src/content/development/contribution-guide/_index.en.md:157]: > Let me give an example. Imagine you were to send a PR creating docs that "Janki decides everything regarding planning", I as a website Owner +2 it, some non-Owner +1s, and we merge it. That would not be an enforceable Submariner policy, because our governance gives Project Owners that responsibility, not the three of us.

This is a disingenuous example as we're actually discussing these policies in the team calls and we're not merging them based on 2 approvals alone.

But, if we were to discuss "Janki decides everything regarding planning" on the calls and agree on it, and sent documentation for it which has a not insignificant majority of the team +1 it, then it would be a viable policy.

Comment (dfarrell07) [reply src/content/development/contribution-guide/_index.en.md:157]: > which has a not insignificant majority of the team +1 it, then it would be a viable policy.

No, this is not true. The "team" (how is that defined?) and a "not insignificant majority" (what does that mean?) are not how our governance allocates decision making authority. Only a majority of Project Owners have the authority to [set policies like this](https://submariner.io/community/contributor-roles/#owner-responsibilities-and-privileges).

**Review by Jaanki — APPROVED** [review]
